sympy/matrices/determinant.py

```diff
@@ -329,12 +328,24 @@ def _adjugate(M, method="berkowitz"):
     return M.cofactor_matrix(method=method).transpose()
 
 
-def _charpoly_DOM(DOM, x, domain):
-    return PurePoly(DOM.charpoly(), x, domain=domain)
+def _charpoly_DOM(M, x, simplify):
+    DOM = DomainMatrix.from_Matrix(M)
+    domain = DOM.domain
+
+    simplify_poly = False
+    if domain.is_FractionField and not all(t.is_Symbol for t in domain.symbols):
+        simplify_poly = True
+
+    x = uniquely_named_symbol(x, M, modify=lambda s: '_' + s)
+    p = PurePoly(DOM.charpoly(), x, domain=domain)
+    if simplify_poly:
+        return PurePoly(simplify(p.as_expr()), x)
```

Comment:
I think it would be better to move this `simplify` and `uniquely_named_symbol` logic into the main `_charpoly` function.

Calling `uniquely_named_symbol` on a large matrix can be slow so I would suggest to create the `Poly` with a `Dummy` symbol and then call `uniqely_named_symbol` on the `Poly`. For DomainMatrix we can actually just callect the symbols from the domain e.g. `ZZ` has no symbols.

Reply:
This function should just return a list of coefficients same as `berk_vector` does.

Reply:
Sure, I have implemented it as instructed.
@oscarbenjamin, kindly review the commit.
Thanks.